subs/pantry/src/Pantry/Repo.hs

```diff
@@ -176,23 +173,21 @@ withRepo repo@(Repo url commit repoType' _subdir) action =
               , ["update", "-C", T.unpack commit]
               , Nothing
               )
+        fixANSIForWindows =
+          -- On Windows 10, an upstream issue with the `git clone` command means that
+          -- command clears, but does not then restore, the
+          -- ENABLE_VIRTUAL_TERMINAL_PROCESSING flag for native terminals. The
+          -- folowing hack re-enables the lost ANSI-capability.
+          when osIsWindows $ void $ liftIO $ hSupportsANSIWithoutEmulation stdout
 
     logInfo $ "Cloning " <> display commit <> " from " <> display url
-    runCommand ("clone" : [T.unpack url, suffix])
-    -- On Windows 10, an upstream issue with the `git clone` command means that
-    -- command clears, but does not then restore, the
-    -- ENABLE_VIRTUAL_TERMINAL_PROCESSING flag for native terminals. The
-    -- folowing hack re-enables the lost ANSI-capability.
-    when osIsWindows $ void $ liftIO $ hSupportsANSIWithoutEmulation stdout
+    runCommand ["clone", T.unpack url, dir]
```

Comment:
Please add a comment that this is intentionally _not_ run within the `withWorkingDir` below in order to allow relative paths to work.